Sage/carbon
# fix(icon): fix tooltip visible logic when disabled

Fixes: #4745

### Proposed behaviour

Fix small bug in `Icon` to prevent the tooltip being permanently disabled when `Icon` is disabled then reenabled.

The raised issue was about icon button, but really this is an `Icon` bug, it can be replicated on the `Icon` test story using the disabled controls.

### Current behaviour

When the `Icon` is disabled it sets the Tippy tooltip to not visible and it can't be re-enabled (without refreshing the page).

### Checklist




#### QA


### Additional context



### Testing instructions

Test the sandbox for the issue below.

The issue can also be replicated on the Icon test story: https://carbon.sbc-ui.com/?path=/story/icon-test--default-story
- Use the `tooltipMessage` control to add a tooltip, then you can hover over the icon to see it
- Disable the icon with the `disabled` control, then re-enable it.
- The tooltip no longer appears on hover.

This will be fixed in the version introduced in this PR
